fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/OrdinalFieldValue.java

```diff
@@ -94,11 +96,19 @@ public String explain(@Nonnull final Formatter formatter) {
     @Nullable
     @Override
     public <M extends Message> Object eval(@Nonnull final FDBRecordStoreBase<M> store, @Nonnull final EvaluationContext context) {
-        final Message childMessage = (Message)child.eval(store, context);
-        if (childMessage == null) {
+        final var result = child.eval(store, context);
+        if (result == null) {
             return null;
         }
-        return MessageValue.getFieldOnMessage(childMessage, field.getFieldIndex());
+        if (result instanceof QueryResult) {
```

Comment:
This cannot happen as discussed. This is always a message.

Reply:
reverted.